dirty_cat/_fuzzy_join.py

```diff
@@ -0,0 +1,176 @@
+"""
+Fuzzy joining tables using string columns.
+The principle is as follows:
+  1. We transform the key columns strings to vectors using CountVectorizer.
+  2. For each category, we use the nearest neighbor method to find its closest
+  neighbor and establish a match.
+  3. We match the tables using the previous information.
+Categories from the two tables that share many n-grams have greater
+probability of beeing matched together. The join is based on
+morphological similarities between strings.
+"""
+
+import pandas as pd
+import numpy as np
+from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
+from sklearn.neighbors import NearestNeighbors
+
+
+def fuzzy_join(left_table, right_table, on, return_distance=False, analyzer="char_wb",
+              ngram_range=(2, 4), precision='nearest', precision_threshold=0.5, suffixes=('_l', '_r'), keep='all',):
```

Comment:
Please type hint your function signatures and return types as much as possible :)